src/meta/partition_guardian.cpp

```diff
@@ -0,0 +1,50 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "partition_guardian.h"
+namespace dsn {
+namespace replication {
+partition_guardian::partition_guardian(meta_service *svc) : _svc(svc) {}
+
+bool partition_guardian::collect_replica(meta_view view,
+                                         const rpc_address &node,
+                                         const replica_info &info)
+{
+    partition_configuration &pc = *get_config(*view.apps, info.pid);
```

Comment:
It's better to refactor `get_config` and change the raw pointer to a shared pointer

Reply:
I don't want to modify any code in this pull request. My purpose is to remove simple_load_balancer eventually.

Reply:
is `get_config` abandoned in the future?

Reply:
I have no idea. My purpose is to moving code in simple_load_balancer to partition_guardian